### ISSUE TITLE
Extra leading underscore in type parameter names

### ISSUE THREAD

**Issue by azz** [eslint/typescript-eslint-parser#296]
**What version of TypeScript are you using?**

2.3.2

**What version of `typescript-eslint-parser` are you using?**

`master`

**What code were you trying to parse?**

```ts
class A<__P> {}
```

**What did you expect to happen?**

```js
assert(require('.').parse('class A<__P> {}').body[0].typeParameters.params[0].name === "__P");
```

**What happened?**

The `name` if the parameter is `___P` (notice the extra leading underscore. This continues if there are more than two leading `_` characters (always N+1 reported).

**Comment by soda0289** [comment]
This is a feature of the typescript AST. We remove the extra underscore in Identifiers but have missed this case. Should be easy fix we have function to handle this case.
